lib/internal/Magento/Framework/App/FeedInterface.php

```diff
@@ -0,0 +1,14 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Framework\App;
+
+interface FeedInterface
+{
+    /**
+     * @return string
+     */
+    public function asXml();
```

Comment:
I prefer to see this method as 

```php
public function getFormatedContentAs($format = SupportedFeedFormatsInterface::DEFAULT_FORMAT);
```

Naming could be changed, but main idea is to make FeedInteface agnostic to the possible representation formats.
Because we no need to have coupling between Data and different ways how these data could be presented. 
Thus, all that we need to have is kind of "output" method which accepts a format in which data should be presented. And we need to have a dedicated interface/entity responsible for format registration.
I mentioned   `SupportedFeedFormatsInterface` here